lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -145,6 +145,68 @@ class MatmulOpConversionPattern : public OpConversionPattern<ttir::MatmulOp> {
 };
 // ANCHOR_END: adding_an_op_matmul_op_rewriter
 
+class Conv2dOpConversionPattern : public OpConversionPattern<ttir::Conv2dOp> {
+public:
+  using OpConversionPattern<ttir::Conv2dOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ttir::Conv2dOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    // auto kernel_ty = mlir::cast<RankedTensorType>(adaptor.getWeight().getType());
+    // llvm::ArrayRef<std::int64_t> kernel_shape = kernel_ty.getShape();
+
+    auto input_ty = mlir::cast<RankedTensorType>(adaptor.getInput().getType());
+    llvm::ArrayRef<std::int64_t> input_shape = input_ty.getShape();
+
+    auto output_ty = mlir::cast<RankedTensorType>(adaptor.getOutput().getType());
+    llvm::ArrayRef<std::int64_t> output_shape = output_ty.getShape();
+
+
+    auto in_channels = rewriter.getI32IntegerAttr(input_shape[input_shape.size()-1]);
+    auto out_channels = rewriter.getI32IntegerAttr(output_shape[output_shape.size()-1]);
+    auto batch_size = rewriter.getI32IntegerAttr(input_shape[input_shape.size()-4]);
```

Comment:
Input rank is not guaranteed to be 4, is it? So if it is 3 you need to handle it here, and also if it is greater than 4 then you should collapse redundant dims into batch size.